app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,103 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+
+    if @task.completed
+      @status = "completed"
+    elsif @task.due_date && @task.due_date < DateTime.now
+      @status = "overdue"
+    else
+      @status = "in-progress"
+    end
+
+    if @task[:completed]
```

Comment:
You should use `@task.completed` at this point instead of square brackets, because `@task` is an object (not a hash). Rails is just weird for letting this syntax be valid.